File: projects/agentic-automation-framework/src/channels/whatsapp.js
export async function sendWhatsAppMessage({ token, phoneNumberId, to, text }) {
  if (!token || !phoneNumberId || !to) return;
  const url = `https://graph.facebook.com/v21.0/${phoneNumberId}/messages`;
  const res = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`
    },
    body: JSON.stringify({
      messaging_product: "whatsapp",
      to,
      type: "text",
      text: { body: text }
    })
  });
  const data = await res.json();
  if (!res.ok) throw new Error(`WhatsApp send failed: ${JSON.stringify(data)}`);
  return data;
}

File: projects/agentic-automation-framework-python/agentic/channels/whatsapp.py
import requests


def send_message(token: str, phone_number_id: str, to: str, text: str):
    if not token or not phone_number_id or not to:
        return None
    url = f"https://graph.facebook.com/v21.0/{phone_number_id}/messages"
    payload = {
        "messaging_product": "whatsapp",
        "to": to,
        "type": "text",
        "text": {"body": text},
    }
    r = requests.post(url, headers={"Authorization": f"Bearer {token}", "Content-Type": "application/json"}, json=payload, timeout=30)
    if not r.ok:
        raise RuntimeError(f"WhatsApp failed: {r.text}")
    return r.json()
